src/Carousel.tsx

```diff
@@ -202,25 +242,27 @@ const Carousel = React.forwardRef((uncontrolledProps, ref) => {
 
   const prefix = useBootstrapPrefix(bsPrefix, 'carousel');
 
-  const nextDirectionRef = useRef(null);
+  const nextDirectionRef = useRef<string | null>(null);
   const [direction, setDirection] = useState('next');
 
   const [isSliding, setIsSliding] = useState(false);
-  const [renderedActiveIndex, setRenderedActiveIndex] = useState(activeIndex);
+  const [renderedActiveIndex, setRenderedActiveIndex] = useState<number>(
+    activeIndex || 0,
```

Comment:
a little concerned with the coercing to `0` having unintended consequences

Reply:
I guess, yeah, but there are places doing arithmetic like `renderedActiveIndex - 1` without checking whether it is might happen to be `undefined`... What do you think, fix those to assume things or..?